### PR TITLE
Enable padded video inputs in the Qwen3* vision encoder

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

This PR wires padded-video metadata through the Qwen3* vision encoder:

- Adds `token_mask` and `valid_grid` support (when provided) to Qwen3* vision RoPE.
- Computes MRoPE cos/sin from the unpadded `valid_grid`, then scatters those values back onto valid token positions in the padded sequence.
- Propagates the patch-level video mask through vision attention so valid tokens do not attend to padded tokens.
- Preserves patch-major video padding layout in the Qwen3-Omni processor.

# Tests

Unit tests `test_padded_video_valid_outputs_match_unpadded` verified locally and in [github CI](https://screenshot.googleplex.com/68oseGSKjLYeXW4):

```
python -m pytest tests/unit/qwen3_omni_layers_test.py::TestQwen3OmniMoeVisionEncoderEndToEnd::test_padded_video_valid_outputs_match_unpadded -vv -s
python -m pytest tests/unit/qwen3_omni_layers_test.py::TestQwen3OmniMoeVisionPatchEmbed::test_patch_embed_padded_video_valid_outputs_match_unpadded -vv -s
```

Example test flow:

`raw_grid = (2, 2, 2)`, `max_grid = (3, 4, 4)`, `tps = 2`, `patch = 16`, `hidden = 16`.

`raw_video`: `(1, 3, 4, 32, 32)` -> `8` patch tokens.

`padded_video`: `(1, 3, 6, 64, 64)` -> `48` patch tokens, where only the first `8` are valid.

`video_mask`: `(1, 1, 6, 64, 64)` is reduced inside the encoder to patch-level `attention_mask`: `(1, 48)`.

For padded input, `video_grid_thw = raw_grid = (2, 2, 2)` becomes `valid_grid`.

The encoder uses:
- padded static grid `(3, 4, 4)` for sequence shape,
- **`valid_grid = (2, 2, 2)` to compute valid pos embeddings and MRoPE,**
- **`token_mask = attention_mask` to scatter valid pos/RoPE values into the padded sequence and block attention to padded tokens.**

The test compares:

`raw_output.reshape(8, 16)` vs `padded_output[patch_mask]`, also `(8, 16)`.

They match within tolerance, proving padded tokens do not affect valid vision encoder outputs for this example.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/models/qwen3.py [hunk 1]
```diff
@@ -1959,11 +1963,14 @@ def __call__(
...
         "height": height,
         "width": width,
+        "token_mask": attention_mask,
+        "valid_grid": valid_grid,
     }
     output, _ = self.attn(
         inputs_q=hidden_states,
         inputs_kv=hidden_states,
         deterministic=deterministic,
+        decoder_segment_ids=attention_mask,
```
Comment (hengtaoguo) [inline src/maxtext/models/qwen3.py:1973]: Only valid visual tokens can attend to each other in the vision encoder.

File: src/maxtext/layers/encoders.py [hunk 2]
```diff
@@ -93,10 +93,15 @@ def _setup_vision_encoder_layers(self):
     else:
       raise ValueError(f"No VisionEncoder implemented for {self.config.model_name} yet")
 
-  def __call__(self, input_images, deterministic=False):
+  def __call__(self, input_images, input_masks=None, video_grid_thw=None, deterministic=False):
```
Comment (aireenmei) [inline src/maxtext/layers/encoders.py:96]: AI suggested that the call sites of this function (self.vision_encoder(...) in `src/maxtext/models/models.py`) don't have in_masks or video_grid_thw passed. Do you plan to add later?

**Review by aireenmei — APPROVED** [review]
Can we run end-to-end decode test? Can be in future PR

File: src/maxtext/layers/encoders.py (hunk 2 above)
Comment (hengtaoguo) [reply src/maxtext/layers/encoders.py:96]: Thanks for the catch! This PR still hasn't supported E2E decode yet, and it's promising to do that in the next immediate PR (wire inputs all the way, and also connector).

After that, we will be comfortable processing batched input with static video tensor.

**Review by abhinavclemson — APPROVED** [review]
